### ISSUE TITLE
no-unused-variable finds false positives in imports

### ISSUE THREAD

**Issue by aj-r** [palantir/tslint#3854]
### Bug Report

- __TSLint version__: 5.9.1
- __TypeScript version__: 2.8.3
- __Running TSLint via__: CLI

#### TypeScript code being linted

`animal.ts`:
```ts
export interface Dog {
    type: "dog";
    ...
}
export interface Cat {
    type: "cat";
    ...
}
export type Animal = Dog | Cat;
```

`otherFile.ts`:

```ts
import { Animal, Cat, Dog } from "./animal"

export const checkAnimal = (animal: Animal | undefined): void => {
    ...
}
```

with `tslint.json` configuration:

```json
{
  "extends": [
    "tslint:latest",
    "tslint-eslint-rules"
  ],
  "rules": {
    "no-unused-variable": true
  }
}
```

#### Actual behavior

```
ERROR: 1:1   no-unused-variable        'Cat' is declared but its value is never read.
ERROR: 1:1   no-unused-variable        'Dog' is declared but its value is never read.
```

Note that importing `Cat` and `Dog` is required. If I omit them from the imports, I get a typescript error:

```
ERROR in [...]/otherFile.ts
./otherFile.ts
[tsl] ERROR in [...]/otherFile.ts(170,5)
      TS4029: Exported function 'checkAnimal' of exported class has or is using name 'Cat' from external module "[...]/animal" but cannot be named.
```

This particular case might be a typescript bug, but I believe there are similar cases (especially implicitly typed variables) where this issue still applies.

#### Expected behavior

No error.

This is different than #3463.

**Comment by suchanlee** [comment]
See related discussion: https://github.com/palantir/tslint/pull/3853#issuecomment-384540618

**Comment by suchanlee** [comment]
It's unlikely that we'll actively support this rule in the future and will encourage using the typescript flag
